src/segy/header.py

```diff
@@ -0,0 +1,194 @@
+"""Header interface.
+
+We use facade structural design pattern for HeaderAccessor and strategy
+patterns to make header interactions more user-friendly and configurable.
+"""
+
+from __future__ import annotations
+
+from json import dumps as json_dumps
+from typing import TYPE_CHECKING
+
+from pandas import DataFrame
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    from numpy import dtype
+    from numpy.typing import NDArray
+
+    from segy.schema import Endianness
+
+
+class HeaderTransformationStrategy:
+    """Base class for header transformation strategies."""
+
+    def transform(self, data: NDArray[Any]) -> NDArray[Any]:
+        """Forward transformation implementation."""
+        raise NotImplementedError
+
+    def inverse_transform(self, data: NDArray[Any]) -> NDArray[Any]:
+        """Inverse transformation implementation."""
+        raise NotImplementedError
+
+
+class ScalingStrategy(HeaderTransformationStrategy):
+    """Transform to scale user defined keys in a structured dtype.
+
+    Args:
+        scale_factor: Scalar to use in transformations.
+        keys: List of keys to apply the transformation on.
+    """
+
+    def __init__(self, scale_factor: int | float, keys: list[str]):
+        self.scale_factor = scale_factor
+        self.keys = keys
+
+    def transform(self, data: NDArray[Any]) -> NDArray[Any]:
+        """Forward transformation implementation."""
+        if data.dtype.names is None:
+            msg = f"{self.__class__.__name__} can only work on structured arrays."
+            raise ValueError(msg)
+
+        for key in self.keys:
+            if key in data.dtype.names:
+                data[key] = [vals * self.scale_factor for vals in data[key]]
```

Comment:
This could be handled when creating the object, but I believe the scaling operation depends on if the scalar is positive or negative. When positive it's a multiplication and when negative it's a division by abs(scalar).

Reply:
I mainly left the scaling part generic, so the logic for negative and positive numbers must be handled outside and passed as scale coefficient here. Just wanted to keep it flexible so if you want to scale something with an arbitrary number, it still works.